utils/prezly/api/PrezlyApi.ts

```diff
@@ -67,43 +73,64 @@ export default class PrezlyApi {
         return stories;
     }
 
-    async getStories(limit = DEFAULT_STORIES_COUNT, order: SortOrder = DEFAULT_SORT_ORDER) {
+    async getStories({
+        page = 0,
+        pageSize = DEFAULT_PAGE_SIZE,
+        order = DEFAULT_SORT_ORDER,
+    }: GetStoriesOptions = {}) {
         const sortOrder = getSortByPublishedDate(order);
         const jsonQuery = JSON.stringify(getStoriesQuery(this.newsroomUuid));
 
-        const { stories } = await this.searchStories({ limit, sortOrder, jsonQuery });
-        return stories;
-    }
+        const { stories, pagination } = await this.searchStories({
+            limit: pageSize,
+            offset: page ? (page - 1) * pageSize : undefined,
+            sortOrder,
+            jsonQuery,
+        });
 
-    async getStoriesExtended(limit = DEFAULT_STORIES_COUNT, order: SortOrder = 'desc') {
-        const stories = await this.getStories(limit, order);
-        const extendedStoriesPromises = stories.map((story) => this.getStory(story.id));
+        const storiesTotal = pagination.matched_records_number;
 
-        return Promise.all(extendedStoriesPromises);
+        return { stories, storiesTotal };
     }
 
-    async getStoriesExtendedFromCategory(
-        category: Category,
-        limit = DEFAULT_STORIES_COUNT,
-        order: SortOrder = DEFAULT_SORT_ORDER,
-    ) {
-        const stories = await this.getStoriesFromCategory(category, limit, order);
-        const extendedStoriesPromises = stories?.map((story) => this.getStory(story.id)) || [];
+    async getStoriesExtended(options?: GetStoriesOptions) {
+        const { stories } = await this.getStories(options);
+        const extendedStoriesPromises = stories.map((story) => this.getStory(story.id));
 
         return Promise.all(extendedStoriesPromises);
     }
 
     async getStoriesFromCategory(
         category: Category,
-        limit = DEFAULT_STORIES_COUNT,
-        order: SortOrder = DEFAULT_SORT_ORDER,
+        {
+            page = 0,
+            pageSize = DEFAULT_PAGE_SIZE,
+            order = DEFAULT_SORT_ORDER,
+        }: GetStoriesOptions = {},
     ) {
         const sortOrder = getSortByPublishedDate(order);
         const jsonQuery = JSON.stringify(getStoriesQuery(this.newsroomUuid, category.id));
 
-        const { stories } = await this.searchStories({ limit, sortOrder, jsonQuery });
+        const { stories, pagination } = await this.searchStories({
+            limit: pageSize,
+            offset: page ? (page - 1) * pageSize : undefined,
```

Comment:
Ok, I'm getting a small headache from this :)
Which page is first, `0` or `1`?

If `1`, then the default value of `page` should not be `0` - it's not allowed, so it's confusing why it's (suddenly) there.
It's better to use a special value (like `undefined`) to handle a default page value. 

```suggestion
            offset: typeof page === 'undefined' ? undefined : (page - 1) * pageSize,
```

Reply:
My bad, I was starting with zero-index, and the switched to this, but forgot to update the parameter 🤦🏻